src/Integration/Connection/ConnectionWorkflow.cs

```diff
@@ -183,6 +183,16 @@ private void OnProjectsChanged(ConnectionInformation connection, IEnumerable<Son
                 var projects = await this.host.SonarQubeService.GetAllProjectsAsync(connection.Organization?.Key,
                     cancellationToken);
 
+                if (string.IsNullOrEmpty(connection.Password.ToUnsecureString()) &&
+                    projects.Count == 10000 &&
+                    !projects.Any(p => p.Key == this.host.VisualStateManager.BoundProjectKey))
+                {
+                    projects = new List<SonarQubeProject>
+                    {
+                        new SonarQubeProject(this.host.VisualStateManager.BoundProjectKey, this.host.VisualStateManager.BoundProjectName)
```

Comment:
We need the project name otherwise the selected/bound project doesn't show-up correctly.